src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -10490,6 +10490,115 @@ public void SynthesizedDelegateTypes_NamedArguments_MoreThan16Parameters()
                 Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "inferred").WithArguments("arg1", "<anonymous delegate>").WithLocation(7, 1));
         }
 
+        [Fact]
+        public void SynthesizedDelegateTypes_MoreThan16Parameters_DefaultParameterValue()
+        {
+            var range = Enumerable.Range(1, 17);
+            var manyParams = string.Join(", ", range.Select(i => $"int p{i}"));
+            var manyTypes = string.Join(",", range.Select(_ => "System.Int32"));
+            var source = $$"""
+                using System;
+                static void Report(Delegate d) => Console.WriteLine(d.GetType());
+                var lam1 = ({{manyParams}} = 1) => { };
+                Report(lam1);
+                var lam2 = ({{manyParams}} = 1) => { };
+                Report(lam2);
+                var lam3 = ({{manyParams}} = 2) => { };
+                Report(lam3);
+                """;
+            CompileAndVerify(source, expectedOutput: $"""
+                <>f__AnonymousDelegate0`17[{manyTypes}]
```

Comment:
><>f__AnonymousDelegate0`17[{manyTypes}] [](http://example.com/codeflow?start=16&length=39)

I am not sure what is the point making type of p17 a generic type parameter. Can default value ``` = 1``` be unified with any type other than System.Int32?

Reply:
Perhaps default value can be unified between ```int``` and ```int?```. Let's demonstrate that, if that is the case.

Reply:
I guess this does have nothing to do with this particular test (for more than 16 parameters), but I will demonstrate what you suggest in another test. Thanks.